File: src/main/java/com/brainate/domain/Statistic.java
package com.brainate.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "statistics")
@NoArgsConstructor
@Data
@AllArgsConstructor
public class Statistic {

    @Id
    @Column(name = "id_statistic")
    @SequenceGenerator(name = "statistic_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "statistic_seq")
    @NotNull
    private Long id;

    @Column(name = "mode")
    @Enumerated(EnumType.STRING)
    private Mode mode;

    @Column(name = "added")
    private Long added;

    @Column(name = "done")
    private Long done;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_language")
    @JsonBackReference
    private Language language;
}

File: src/client/src/app/model/Statistic.ts
import {Language} from "./Language";
import {Mode} from "./Mode";

export class Statistic {
  id: number;
  mode: Mode;
  added: number;
  done: number;
  language: Language;

  constructor(mode?: Mode, added?: number, done?: number, language?: Language) {
    this.mode = mode;
    this.added = added;
    this.done = done;
    this.language = language;
  }

}
